muacrypt/account.py

```diff
@@ -327,6 +327,53 @@ def process_incoming(self, msg):
             account=self,
         )
 
+    def process_incoming_gossip(self, dec):
+        """ process gossip from encrypted part of incoming mail message
+        and update states information from any gossip header for the
+        To or Cc recipients of the msg.
+
+        :type dec: DecryptMimeResult
+        :param dec: DecryptMimeResult instance as returned from decrypt_mime
+        :rtype: ProcessIncomingResult
+        """
+        print(dec)
+
+        msg = dec
+        msg_date = effective_date(parse_date_to_float(msg.get("Date")))
+        msg_id = six.text_type(msg["Message-Id"])
+        recipients = mime.get_target_emailadr(dec)
+        peerstates = {}
+        pahs = {}
+        for recipient in recipients:
+            peerstate = self.get_peerstate(recipient)
+            r = mime.parse_one_gossip_header_from_msg(msg, recipient)
```

Comment:
so for each recipient address you parse all gossip headers which i find questionable from an algorithmic complexity perspective. 

i rather suggest to parse all gossip headers once and match them with recipients. If there are more than one gossip header for a recipient then drop that. But drop duplicate gossip headers for a recipient  -- i can imagine this to happen if you mail twice to the same addr and the sender has a dumb "add one gossip for each recipient" algorithm. 

Also i suggest to test the ``mime.parse_gossip_headers()`` in ``test_mime.py`` separately (including the dropping of duplicates).